contentcuration/contentcuration/utils/publish.py

```diff
@@ -708,18 +728,40 @@ def process_image_strings(content, zf, channel_id):
                     logging.warning("NOTE: the following error would have been swallowed silently in production")
                     raise
 
-            image_name = "images/{}.{}".format(checksum, ext[1:])
-            if image_name not in zf.namelist():
-                with storage.open(ccmodels.generate_object_storage_name(checksum, filename), 'rb') as imgfile:
-                    write_to_zipfile(image_name, imgfile.read(), zf)
-
-            # Add resizing data
-            if img_match.group(2) and img_match.group(3):
-                image_data = {'name': img_match.group(1)}
-                image_data.update({'width': float(img_match.group(2))})
-                image_data.update({'height': float(img_match.group(3))})
-                image_list.append(image_data)
-            content = content.replace(match.group(1), img_match.group(1))
+            original_image_name = "images/{}.{}".format(checksum, ext[1:])
+            with storage.open(ccmodels.generate_object_storage_name(checksum, filename), 'rb') as imgfile:
```

Comment:
After your changes here, it seems we'll always read the image from storage, for every match in the for-loop that contains this.